server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraSessionModule.java

```diff
@@ -85,6 +87,17 @@ Cluster provideCluster(FileSystem fileSystem, AsyncRetryExecutor executor) throw
                 .get();
     }
 
+    @Provides
+    @Singleton
+    ZonedDateTimeCodec provideZonedDateTimeCodec(Session session, CassandraTypesProvider cassandraTypesProvider) {
```

Comment:
It does much more than providing the codec, it also registers it. You should do only one thing at a time.